### PR TITLE
Fix `mcmb` on IBM i (OS/400) PASE; fix DMD ImportC 2.107.0 incompatibility

### PR DESCRIPTION
**Pull request by johnsonjh**
* Fix IBM i (OS/400) PASE `mcmb` compatibility
* Update to DMD ImportC 2.107.0 and workaround a bug

### CODE REVIEW

File: src/sirinternal.c [hunk 1]
```diff
@@ -55,10 +55,17 @@ static sirconfig _sir_cfg      = {0};
 static sirfcache _sir_fc       = {0};
 static sir_plugincache _sir_pc = {0};
 
+#if !defined(__IMPORTC__)
 static sir_mutex cfg_mutex  = SIR_MUTEX_INIT;
 static sir_mutex fc_mutex   = SIR_MUTEX_INIT;
 static sir_mutex pc_mutex   = SIR_MUTEX_INIT;
 static sir_mutex ts_mutex   = SIR_MUTEX_INIT;
+#else
```
Comment (aremmell) [inline src/sirinternal.c:63]: just do this where SIR_MUTEX_INIT is defined, silly.

File: tests/tests.c [hunk 2]
```diff
@@ -1099,7 +1099,11 @@ bool sirtest_mutexsanity(void) {
     TEST_MSG_0(WHITEB("create, lock, unlock, destroy"));
     (void)printf(INDENT_ITEM WHITE("creating mutex...") "\n");
 
+#if !defined(__IMPORTC__)
     sir_mutex m1 = SIR_MUTEX_INIT;
+#else
```
Comment (aremmell) [inline tests/tests.c:1104]: Same

**Review by aremmell — CHANGES_REQUESTED** [review]

File: src/sirinternal.c (hunk 1 above)
Comment (johnsonjh) [reply src/sirinternal.c:63]: That's the bug tho, as discussed.

File: tests/tests.c (hunk 2 above)
Comment (johnsonjh) [reply tests/tests.c:1104]: =)

**Review by aremmell — APPROVED** [review]

**Pull request merged**